cfme/tests/test_db_restore.py

```diff
@@ -0,0 +1,120 @@
+# -*- coding: utf-8 -*-
+
+import fauxfactory
+import pytest
+
+from cfme.configure import configuration as config
+from cfme.cloud import provider as cloud_provider
+from cfme.infrastructure.provider import wait_for_a_provider
+from cfme.infrastructure.virtual_machines import Vm
+from fixtures.pytest_store import store
+from utils.appliance import provision_appliance
+from utils.log import logger
+from utils.providers import setup_a_provider
+from utils import testgen, version
+
+pytest_generate_tests = testgen.generate(testgen.infra_providers, scope="module")
+
+
+def pytest_generate_tests(metafunc):
+    argnames, argvalues, idlist = testgen.provider_by_type(
+        metafunc, ['virtualcenter'], choose_random=True)
+    if not idlist:
+        return
+    metafunc.parametrize(argnames, argvalues, ids=idlist, scope='module')
+
+
+@pytest.fixture(scope="module")
+def test_vm(request, provider_crud, provider_mgmt):
+    """Fixture to provision appliance to the provider being tested if necessary"""
+    vm_name = "test_rest_db_" + fauxfactory.gen_alphanumeric()
+    vm = Vm(vm_name, provider_crud)
+
+    request.addfinalizer(vm.delete_from_provider)
+
+    if not provider_mgmt.does_vm_exist(vm_name):
+        logger.info("deploying {} on provider {}".format(vm_name, provider_crud.key))
+        vm.create_on_provider(allow_skip="default")
+    else:
+        logger.info("recycling deployed vm {} on provider {}".format(vm_name, provider_crud.key))
+    vm.provider_crud.refresh_provider_relationships()
+    vm.wait_to_appear()
+    return vm
+
+
+@pytest.fixture(scope="module")
+def get_appliances():
+    """Returns two database-owning appliances
+
+    """
+    ver_to_prov = str(version.current_version())
+    appl1 = provision_appliance(ver_to_prov, 'test_back')
+    appl2 = provision_appliance(ver_to_prov, 'test_rest')
+    appl1.configure(region=0, patch_ajax_wait=False)
+    appl1.ipapp.wait_for_web_ui()
+    appl2.configure(region=0, patch_ajax_wait=False)
+    appl2.ipapp.wait_for_web_ui()
+    return (appl1, appl2)
+
+
+@pytest.mark.uncollectif(
+    lambda: not (store.current_appliance.version > '5.4' and
```

Comment:
This will only test on 5.5 and above:
`store.current_appliance.version > '5.4'`, need to tweak the comparator to include 5.4. Also we can put the downstream check before the version check to short circuit quickly on upstream apps:

`lambda: not (store.current_appliance.is_downstream and store.current_appliance.version >= '5.4')`
So "Collect this test if the current appliance is downstream, and if the current version is 5.4 or higher"